app/src/debug/java/org/mozilla/fenix/DebugFenixApplication.kt

```diff
@@ -8,20 +8,31 @@ import android.os.StrictMode
 import androidx.preference.PreferenceManager
 import leakcanary.AppWatcher
 import leakcanary.LeakCanary
+import org.mozilla.fenix.ext.application
 import org.mozilla.fenix.ext.getPreferenceKey
 
 class DebugFenixApplication : FenixApplication() {
 
     override fun setupLeakCanary() {
+        if (!AppWatcher.isInstalled) {
+            AppWatcher.manualInstall(
+                application = application,
+                watchersToInstall = AppWatcher.appDefaultWatchers(application)
```

Comment:
Since you're just using the default set up, you could rely on kotlin default params instead: `AppWatcher.manualInstall(application)`

One question: why do you need to to check `if (!AppWatcher.isInstalled) {` ?  Are you not disabling auto install on all build flavors? Or is this because `setupLeakCanary()` can be called several times per Application lifetime?

Reply:
@pyricau thanks for taking a look at this. 

The check is just a precautionary measure. The setup method is implemented just in the debug app. 

For the default param, I chose to state them explicitly, in case we later want to filter the watchers we want to use.